app/src/main/java/com/example/taskyy/ui/LoginScreen.kt

```diff
@@ -0,0 +1,166 @@
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.AbsoluteRoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.taskyy.R
+
+@Composable
+fun MainScreen() {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.SpaceAround,
+        modifier = Modifier
+            .fillMaxHeight()
+            .background(Color.Black)
+    ) {
+        AddWelcomeBackText()
+        Surface(
+            modifier = Modifier
+                .fillMaxHeight(.80F),
+            shape = AbsoluteRoundedCornerShape(30.dp, 30.dp)
+        ) {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Top,
+                modifier = Modifier
+                    .fillMaxHeight(.75f)
+                    .padding(30.dp)
+            ) {
+                CreateUserNameField()
+                Spacer(modifier = Modifier.height(10.dp))
+                CreatePasswordField()
+                Spacer(modifier = Modifier.height(10.dp))
+                CreateLoginButton()
+                Row(
+                ) {
+                    CreateBottomText()
+                    CreateSignUpLink()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun AddWelcomeBackText() {
+    Text(
+        text = "Welcome Back!",
+        color = Color.White,
+        fontSize = 30.sp,
+        fontWeight = FontWeight.Bold
+    )
+}
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateUserNameField() {
+    var textFieldState by remember {
+        mutableStateOf("")
+    }
```

Comment:
I would always put all UI states in your ViewModel. Using them here with remember will clear the states after a configuration change like a screen rotation

Reply:
I know that using a view model instance in the UI also isn't great for testing so if the state is in the view model, what is the best way to get it to the composable? Is it best to create that view model instance in the activity and then grab the state from the main activity using that instance and pass it as a parameter to the composable? In my case I would pass it to the MainScree() composable?

Reply:
This is what I suggest:
```
// In your ViewModel
@HiltViewModel
class SampleViewModel @Inject constructor(): ViewModel() {
    private val _state = MutableStateFlow(SampleState()) // Alternatively, Compose state also works
    val state = _state.asStateFlow()

    fun onEvent(event: SampleEvent) {
        when(event) {
            is SampleEvent.Event1 -> // Handle event1
            is SampleEvent.Event2 -> // Handle event2
            // ...
        }
    }

    // ...
}

// In your NavHost
composable(Routes.sample) {
    val viewModel = hiltViewModel<SampleViewModel>()
    val state by viewModel.state.collectAsStateWithLifecycle()
    SampleScreen(
        state = state,
        onEvent = viewModel::onEvent
    )
}

// Your screen composable
@Composable
fun SampleScreen(
    state: SampleState,
    onEvent: (SampleEvent) -> Unit
) {
    // ...
    Button(onClick = { onEvent(SampleEvent.ClickedButtonX) } {
        Text("Click me")
    }
}
```
The screen composable where the real UI is located remains totally testable and previewable, since it doesn't depend on a VM instance :)

Reply:
This is my first time working with JetPack Compose and states so pardon the seemingly simple question. You have "SampleState()" being passed as a parameter to the MutableStateFlow(). Where does that come from? I'm trying to implement this but I'm not sure what needs to go there.

Reply:
Where to do you recommend the nav host go? Should that be in the MainActivity or should that go with the LoginScreen Composable file?

Reply:
Separate file, I just put it here for simplicity :)